lib/token-assert.js

```diff
@@ -110,12 +113,50 @@ TokenAssert.prototype.differentLine = function(options) {
         return;
     }
 
-    if (token.loc.end.line === nextToken.loc.start.line) {
+    var atLeast;
+    var atMost;
+    var exactly;
+
+    var useOptions = options.hasOwnProperty('atLeast') ||
+        options.hasOwnProperty('atMost') ||
+        options.hasOwnProperty('exactly');
+
+    if (useOptions) {
+        atLeast = options.atLeast;
+        atMost = options.atMost;
+        exactly = options.exactly;
+    } else {
+        atLeast = 1;
+    }
+
+    var errorMessage = token.value + ' and ' + nextToken.value + ' should be ';
+
+    var emitError = (function(message) {
         this.emit('error', {
-            message: options.message || token.value + ' and ' + nextToken.value + ' should be on different lines',
+            message: options.message || errorMessage + message + ' line(s) apart',
             line: token.loc.end.line,
             column: token.loc.end.column
         });
+    }).bind(this);
+
+    var lineDifference = Math.abs(token.loc.end.line - nextToken.loc.start.line);
+
+    if (atLeast !== undefined) {
```

Comment:
atLeast or exactly could be 0 in which case !0 wouldn't really give us the logic we are looking for. I agree that this set of if/elses could be cleaned up, I'm not convinced a `switch(true)` is the best way to do it. :-)